### ISSUE TITLE
Version 6.0.2 broke ConversationGUIDs

### ISSUE THREAD

**Issue by lunedis** [dotnet/WatsonTcp#281]
While debugging my project I noticed all my conversationGUIDs (for SyncRequests) were zeroed out.

```
Debug: [WatsonTcpClient] synchronous request sent: 00000000-0000-0000-0000-000000000000
 Debug: [WatsonTcpServer] client [a37039c9-fd04-4307-9049-d88b22780704|127.0.0.1:34966] attempting to register GUID 1d758355-8ef9-4008-8f0b-34c6099e6459
 Debug: [WatsonTcpServer] updated client GUID from a37039c9-fd04-4307-9049-d88b22780704 to 1d758355-8ef9-4008-8f0b-34c6099e6459
 Client Connected [1d758355-8ef9-4008-8f0b-34c6099e6459|127.0.0.1:34966]
 Debug: [WatsonTcpServer] [1d758355-8ef9-4008-8f0b-34c6099e6459|127.0.0.1:34966] synchronous request received: 00000000-0000-0000-0000-000000000000
 Debug: [WatsonTcpClient] synchronous response received: 00000000-0000-0000-0000-000000000000
 Info: [WatsonTcpClient] connecting to 127.0.0.1:13014
 Debug: [WatsonTcpServer] accepted connection from [990790b8-ca3f-419e-9a9a-02ea25459c64|127.0.0.1:34972]
 Debug: [WatsonTcpServer] starting data receiver for [990790b8-ca3f-419e-9a9a-02ea25459c64|127.0.0.1:34972]
```

I think it was this change: https://github.com/dotnet/WatsonTcp/commit/0a06c194ca17faa3d04827ea9ee104e835a9d0f8#diff-7c91816241da5fd1a4158907d17c7c43e00f1989503be901d170a81971d5fcafR87

Downgrading to version 6.0.1 does not actually fix my problem (it is most likely my own code, or an interaction with Unity), but I think conversation guids should not be all zeros?

Thanks for the awesome work on this project!

**Comment by jchristn** [comment]
Hi @lunedis thanks for your kind words and sorry for the delay.  I'm reverting this right now.

**Comment by jchristn** [comment]
Published in NuGet v6.0.6
